stl/inc/xutility

```diff
@@ -4523,16 +4519,11 @@ template <class _Iter1, class _Iter2>
     requires (!sized_sentinel_for<_Iter1, _Iter2>)
 constexpr bool disable_sized_sentinel_for<move_iterator<_Iter1>, move_iterator<_Iter2>> = true;
 
-_EXPORT_STD struct unreachable_sentinel_t;
-namespace _Unreachable_sentinel_detail {
-    struct _Base {
-        template <weakly_incrementable _Winc>
-        _NODISCARD friend constexpr bool operator==(const unreachable_sentinel_t&, const _Winc&) noexcept {
-            return false;
-        }
-    };
-} // namespace _Unreachable_sentinel_detail
-_EXPORT_STD struct unreachable_sentinel_t : _Unreachable_sentinel_detail::_Base {}; // TRANSITION, /permissive-
+_EXPORT_STD struct unreachable_sentinel_t {
+    _NODISCARD constexpr bool operator==(const weakly_incrementable auto&) const noexcept {

```

Comment:
There are some deleted `operator()` abbreviated function templates in `<ranges>` that take plain `auto&&`, but I think this would be the first application of a type-constraint on the parameter of an abbreviated function template. Nothing says "I truly don't care about this parameter" like not naming it _and_ not naming its type. I wouldn't think twice about it in C++23 code, but that small added bit of clarity is not worth possibly regressing C++20. I'll unabbreviate this template.